pkg/admission/admission_test.go

```diff
@@ -313,7 +313,12 @@ func TestAlertmanagerConfigConversion(t *testing.T) {
 }
 
 func api() *Admission {
-	a := New(level.NewFilter(log.NewLogfmtLogger(log.NewSyncWriter(os.Stdout)), level.AllowNone()))
+	a := New(
+		slog.New(slog.NewTextHandler(os.Stdout, &slog.HandlerOptions{
+			// slog level math.MaxInt means no logging
+			Level: slog.Level(math.MaxInt),
```

Comment:
@ArthurSens testing this code and trying to log nothing I found that the approach that I pushed [here](https://github.com/prometheus-operator/prometheus-operator/blob/main/internal/log/slog.go#L101-L104) is not working properly. 

[To disable the logs we need to use math.MaxInt instead of -1](https://stackoverflow.com/questions/16895651/how-to-implement-level-based-logging-in-golang) 

WDYT if we first merge these changes here and after create a different PR to fix LevelNone? [`slog.go`](https://github.com/prometheus-operator/prometheus-operator/blob/main/internal/log/slog.go)